### DISCUSSION TITLE
Error Page + Automatic Reload?

### DISCUSSION THREAD

**Discussion by robacarp** [luckyframework/lucky#1745] [Feature requests & Ideas]
Should the Exception reporting error page attempt to connect to the automatic reloading engine, and auto-reload when an event is triggered?

**Comment by jwoertink** [comment]
That's an interesting thought :thinking:  So like, you're on the error page itself, and then you make a file change, that should reload the error page? I would assume that's how it works now, so if it doesn't, then yeah, that should be fixed :+1:

**Reply by robacarp** [reply]
I think so. Consider this mildly contrived scenario:

- If I have a server running
- the browser is open to `/posts/23/comments`
- I modify code which changes the route on that action to `/posts/23/just_the_comments`
- The server recompiles and the browser refreshes
- The browser is now pointing at an error page, and won't reload on subsequent recompiles

Another example, which has been more relevant to me today is fiddling with some code which raises runtime exceptions on a valid action, like preload errors. When the action crashes, the page is replaced by the error page, but when I try to fix it I have to go over to the browser and hit the refresh button like it's 2016!!!

**Reply by jwoertink** [reply]
at least with those steps we can reproduce it and work out a fix

**Comment by robacarp** [comment]
@jwoertink maybe it's also worth considering capturing the compiler output and streaming it to the browser to display it in an error page. I think I've seen this work in Elixir/Phoenix rather well.
